jujuclient/validation.go

```diff
@@ -21,6 +21,9 @@ func ValidateModelDetails(details ModelDetails) error {
 	if details.ModelUUID == "" {
 		return errors.NotValidf("missing uuid, model details")
 	}
+	if details.ModelType == "" {
```

Comment:
This breaks the client when it's talking to an old API version which doesn't return a model type. (e.g. juju models fails like this:

    % juju models
    missing type, model details not valid
    ERROR missing type, model details not valid

Reply:
The api layer is supposed to fill in any missing model type values from older servers. If I have missed a place, that's a bug for sure.